flake.nix

```diff
@@ -38,447 +34,456 @@
+                  default = false;
+                  description = "Whether to enable MEV-Boost";
+                };
+                endpoint = mkOption {
+                  type = types.str;
+                  default = "http://127.0.0.1:18550";
+                  description = "Listening interface for MEV-Boost server";
+                };
+              };
+              datadir = mkOption {
+                type = types.str;
+                description = "Data directory for the databases";
               };
-
-              script = ''${pkgs.lighthouse}/bin/lighthouse bn \
-            --datadir ${options.lighthouse.datadir} \
-            --network mainnet \
-            --http --http-address ${options.lighthouse.endpoint} \
-            --http-allow-origin "*" \
-            --execution-endpoint ${options.lighthouse.exec.endpoint} \
-            --execution-jwt ${options.lighthouse.datadir}/jwt.hex \
-            --builder ${options.lighthouse.mev-boost.endpoint} \
-            --prune-payloads false \
-            --metrics \
-            ${if options.lighthouse.slasher.enable then
-              "--slasher "
-              + " --slasher-history-length " + (toString options.lighthouse.slasher.history-length)
-              + " --slasher-max-db-size " + (toString options.lighthouse.slasher.max-db-size)
-            else "" }
-          '';
-              wantedBy = [ "multi-user.target" ];
             };
+          };
 
-            # firewall
-            networking.firewall = {
-              allowedTCPPorts = [ 9000 ];
-              allowedUDPPorts = [ 9000 ];
-              interfaces."wg0".allowedTCPPorts = [
-                5052 # lighthouse
+          homestakeros = {
+            system = "x86_64-linux";
+            specialArgs = { inherit inputs outputs; };
+            modules = [
+              ./system
+              ./system/ramdisk.nix
+              ./system/formats/netboot-kexec.nix
+              self.nixosModules.homestakeros
+              home-manager.nixosModules.home-manager
+              disko.nixosModules.disko
+              {
+                nixpkgs.overlays = [
+                  ethereum-nix.overlays.default
+                  outputs.overlays.additions
+                  outputs.overlays.modifications
+                ];
+                system.stateVersion = "23.05";
+              }
+            ];
+          };
+        in
+        rec {
+          # filters options recursively
+          # option exports -- accessible through 'nix eval --json .#exports'
+          exports = nixpkgs.lib.attrsets.mapAttrsRecursiveCond
+            (v: ! nixpkgs.lib.options.isOption v)
+            (k: v: v.type.name)
+            homestakeros_options;
+
+          overlays = import ./overlays { inherit inputs; };
+
+          nixosConfigurations = with nixpkgs.lib; {
+            "homestakeros" = nixosSystem (getAttrs [ "system" "specialArgs" "modules" ] homestakeros);
+          } // (with nixpkgs-stable.lib; { });
```

Comment:
Now that we don't use nixos-generators anymore, this can be simplified to `"homestakeros" = nixosSystem homestakeros;`